tests/ConfigurationTest.php

```diff
@@ -83,13 +83,52 @@ public function testLoadInvalidConfigurationTriggersErrors()
      * @param  array|string $expectedResult
      * @return void
      **/
-    public function testGetWithDefaultConfiguration($xpathExpression, $alwaysReturnArray, $expectedResult)
+    public function testGetWithDefaultConfigurationWithoutCache($xpathExpression, $alwaysReturnArray, $expectedResult)
     {
-        $configuration = new Configuration(__DIR__ . '/Resources/configuration/default.xml', __DIR__ . '/Resources/xsd/default.xsd');
+        $configuration = new Configuration(__DIR__ . "/Resources/configuration/default.xml", __DIR__ . "/Resources/xsd/default.xsd", false);
         $configuration->loadConfiguration(null);
 
-        $this->assertEquals($expectedResult, $configuration->get($xpathExpression, $alwaysReturnArray));
-        $this->assertSame($expectedResult, $configuration->get($xpathExpression, $alwaysReturnArray));
+        $this->assertEquals($expectedResult, $configuration->get($xpathExpression, $alwaysReturnArray) );
+        $this->assertSame($expectedResult, $configuration->get($xpathExpression, $alwaysReturnArray) );
+    }
+
+    /**
+     * testGetWithDefaultConfigurationWithCache
+     *
+     * Tests if Configuration::get returns the expected result from the default configuration (with caching)
+     *
+     * @dataProvider provideTestGetWithDefaultConfiguration
+     *
+     * @access public
+     * @param  string       $xpathExpression
+     * @param  boolean      $alwaysReturnArray
+     * @param  array|string $expectedResult
+     * @return void
+     **/
+    public function testGetWithDefaultConfigurationWithCache($xpathExpression, $alwaysReturnArray, $expectedResult)
+    {
+        $configuration = new Configuration(__DIR__ . "/Resources/configuration/default.xml", __DIR__ . "/Resources/xsd/default.xsd");
+        $configuration->loadConfiguration(null);
+
+        $this->assertEquals($expectedResult, $configuration->get($xpathExpression, $alwaysReturnArray) );
+        $this->assertSame($expectedResult, $configuration->get($xpathExpression, $alwaysReturnArray) );
+    }
+
+     /**
+     * testCachedResultsAreValidWhenLoadingNewFileInRuntime
+     *
+     * Confirms that the cache reads from the currenlty loaded configuration
+     *
+     * @access public
+     * @return void
+     **/
+    public function testCachedResultsAreValidWhenLoadingNewFileInRuntime() {
+        $configuration = new Configuration(__DIR__ . '/Resources/configuration/default.xml', __DIR__ . '/Resources/xsd/default.xsd');
+        @$configuration->loadConfiguration(__DIR__ . '/Resources/configuration/invalid.xml');
```

Comment:
I think this is a bit quirky usage (why do I have to call loadConfiguration if I only use a default config), wouldn't it be better to call loadConfiguration with the default config when you try to get some value and there's no dom loaded?